### PR TITLE
FBXLoader: Fixed morph attributes to match base geometry length

### LINKED ISSUE #28378: FBXLoader morphAttributes.position[n].count is less than attributes.position.count
### Description

Hello,
I'm inspecting this model: `https://raw.githubusercontent.com/catalin-enache/threejs-inspector/master/public/models/__free/fbx/Asuna/2/FreeTestAnimations_reexport.fbx` and I observed that `morphAttributes.position[n].count` is less than `attributes.position.count`.

I have a fiddle here: `https://codesandbox.io/p/sandbox/condescending-swanson-4fypcn?file=%2Fsrc%2Findex.js%3A81%2C4-81%2C142`

I'm printing the loaded fbx. Please check the console for these 2 fields in the `AsunaBodyMesh` (the last skinned mesh of the root children).

image

It can be observed that `attributes.position.count` is `86082` and `morphAttributes.position[0].count` is `34902`.

I'm trying to optimize the object by reconstructing new skinned meshes grouped by materialIndex from groups.
I'm extracting slices from attributes , push them into an array for that material then make a new geometry and skinned mesh for that material. I then replace the original mesh with as many meshes as materials the original mesh had.

The problem is that when I try to port the morphAttributes into new skinned meshes, since they have a different count,
I have no ideea how to slice them to match the attributes slices.

I asked here: https://discourse.threejs.org/t/how-are-related-morphattributes-position-arrays-to-attributes-position/65533
maybe my expectation (that the count should be the same between attributes and morphs) is wrong but I understand they should be equal. 

Then I thought probably I should raise this issue here.

### Reproduction steps

Just load the fbx asset and inspect the result: `morphAttributes.position[n].count` and `attributes.position.count`

### Code

no code required

### Live example

`https://codesandbox.io/p/sandbox/condescending-swanson-4fypcn?file=%2Fsrc%2Findex.js%3A81%2C4-81%2C142`

### Screenshots

Provided in description

### Version

0.164.1

### Device

Desktop

### Browser

Edge

### OS

MacOS

### PR DESCRIPTION
**Pull request by catalin-enache**
Fixed #28378

**Description**

Injected `baseVertexPositions` into `geoInfo` so that inside `genFace` we can generate triangles from real geometry,
enforcing  faces that are not involved in morph to be pushed into morphAttributes.

`genFace` is used when generating base geometry as well as in generating morphs.
In `genFace` the `vertices` array is just a mean to generate triangles. 
If we generate `vertices` from `morphPositions`, we wont have triangles for faces not involved in morph,
because `ShapeUtils.triangulateShape` returns an empty array of triangles when fed with for [0,0] elements
That's why we generate them from base geometry.
Later when `buffers.vertex` is being populated, the actual information is retrieved from `morphPositions`.

TLDR: we generate triangles from base geometry and based on these triangles we push data that we read from `morphPositions` into the final morph buffers.

### CODE REVIEW

File: examples/webgl_loader_fbx.html [hunk 1]
```diff
@@ -30,13 +30,29 @@
...
+			const params = {
+				asset: 'Samba Dancing'
+			};
+
+			const assets = [
+				'Samba Dancing',
+				'morph_test'
+			];
+
+			const transforms = {
```
Comment (Mugen87) [inline examples/webgl_loader_fbx.html:49]: Do you think you can update `morph_test.fbx` such that no additional transforms are required in the example? Ideally the asset does not need be scaled and translated after the import. That would simplify the example a bit.

**Comment by Mugen87** [comment]
For testing: https://rawcdn.githack.com/catalin-enache/three.js_fork/FBXLoader_fix_morphs_attributes_length/examples/webgl_loader_fbx.html

File: examples/webgl_loader_fbx.html (hunk 1 above)
Comment (catalin-enache) [reply examples/webgl_loader_fbx.html:49]: Yes I can, however, the existing rabbit is already huge. Should we let it like that ?

Comment (catalin-enache) [reply examples/webgl_loader_fbx.html:49]: I could, re-export the rabbit too to make it have a reasonable size

Comment (Mugen87) [reply examples/webgl_loader_fbx.html:49]: The rabbit is used in other examples I would not like to touch them now. Notice that I have updated your branch and removed the nurbs and rabbit asset from the demo. They are already used in other demos so in `webgl_loader_fbx`  I would only add assets which are not used elsewhere.

Comment (catalin-enache) [reply examples/webgl_loader_fbx.html:49]: I see you removed the rabbit and nurbs. 
Either we don't show it or I can re-export it (I'll do it in blender if so).
Meanwhile I'll just re-export the test example.

Comment (catalin-enache) [reply examples/webgl_loader_fbx.html]: Pushed, changes.

Comment (catalin-enache) [reply examples/webgl_loader_fbx.html]: The CI / E2E testing fail I think it was random. I tested manually that page (webgl_animation_skinning_ik) and looks fine (and it uses a gltf asset anyways).

**Review by Mugen87 — APPROVED** [review]
The example looks good to me!

I'm not really familiar with FBX so I hope @looeee can have a look at the `FBXLoader` related changes. He knows the implementation best.

**Comment by mrdoob** [comment]
The code looks good to me 👌
